src/dask_awkward/lib/core.py

```diff
@@ -1592,6 +1592,185 @@ def _from_iter(obj):
+
+    # Regularise the axis to (0, None)
+    if axis == 0 or axis == -1 * array.ndim:
+        axis = 0
+    elif axis is not None:
+        raise ValueError(axis)
+
+    if combiner is None:
+        combiner = reducer
+
+    if is_positional:
+        assert combiner is reducer
+
+    # For `axis=None`, we prepare each array to have the following structure:
+    #   [[[ ... [x1 x2 x3 ... xN] ... ]]] (length-1 outer lists)
+    # This makes the subsequent reductions an `axis=-1` reduction
+    if axis is None:
+        prepared_array = map_partitions(_prepare_axis_none_chunk, array)
+    else:
+        prepared_array = array
+
+    chunked_fn = _chunk_reducer_non_positional
+    tree_node_fn = _chunk_reducer_non_positional
+    concat_fn = _concat_reducer_non_positional
+    finalize_fn = _finalise_reducer_non_positional
+
+    chunked_kwargs = {
+        "reducer": reducer,
+        "is_axis_none": axis is None,
+        "mask_identity": mask_identity,
+    }
+    tree_node_kwargs = {
+        "reducer": combiner,
+        "is_axis_none": axis is None,
+        "mask_identity": mask_identity,
+    }
+
+    concat_kwargs = {"is_axis_none": axis is None}
+    finalize_kwargs = {
+        "reducer": combiner,
+        "mask_identity": mask_identity,
+        "keepdims": keepdims,
+        "is_axis_none": axis is None,
+    }
+
+    from dask.layers import DataFrameTreeReduction
+
+    token = token or tokenize(
+        array,
+        reducer,
+        label,
+        dtype,
+        split_every,
+        chunked_kwargs,
+        tree_node_kwargs,
+        concat_kwargs,
+        finalize_kwargs,
+    )
+    name_tree_node = f"{label}-tree-node-{token}"
+    name_finalize = f"{label}-finalize-{token}"
+
+    chunked_fn = partial(chunked_fn, **chunked_kwargs)
+    tree_node_fn = partial(tree_node_fn, **tree_node_kwargs)
+    concat_fn = partial(concat_fn, **concat_kwargs)
+    finalize_fn = partial(finalize_fn, **finalize_kwargs)
+
+    if split_every is None:
+        split_every = 8
+    elif split_every is False:
+        split_every = sys.maxsize
+    else:
+        pass
+
+    name_chunked = f"{label}-prepare-{token}"
+    chunked = partitionwise_layer(chunked_fn, name_chunked, prepared_array)
+
+    dftr = DataFrameTreeReduction(
+        name=name_finalize,
+        name_input=name_chunked,
+        npartitions_input=prepared_array.npartitions,
+        concat_func=concat_fn,
+        tree_node_func=tree_node_fn,
+        finalize_func=finalize_fn,
+        split_every=split_every,
+        tree_node_name=name_tree_node,
+    )
+
+    # Build graph
+    prepared_array_graph = prepared_array.__dask_graph__()
+    layers = {name_finalize: dftr, name_chunked: chunked, **prepared_array_graph.layers}
+    dependencies = {
+        name_finalize: {name_chunked},
+        name_chunked: {prepared_array.name},
+        **prepared_array_graph.dependencies,
+    }
+    graph = HighLevelGraph(layers, dependencies)
```

Comment:
@douglasdavis would you be able to cast your eyes over this and check that it is reasonable?

Reply:
yep will do!

Reply:
@agoose77 this is looking good to me.

One thing I want to clarify: It looks to me that you're building the graph with `HighLevelGraph(...)` instead of `HighLevelGraph.from_collections` because `chunked` is a layer instead of a collection. Is there a reason you're making `chunked` via `partitionwise_layer` instead of via `map_partitions` (which will just make a new collection that you can pass to `DataFrameTreeReduction`)?

Reply:
Technically there's no reason for this PR as it stands. I originally needed this because some of the tree reduction steps returned tuples rather than Arrays. I think on-balance I'd remove this, though; the commits in this PR include the previous stab towards positional reducers should we see fit to restore them.